src/main/java/org/apache/commons/text/similarity/LongestCommonSubsequence.java

```diff
@@ -91,65 +309,182 @@ public CharSequence logestCommonSubsequence(final CharSequence left, final CharS
+     * @param rightEnd   End index (inclusive) of the first input subsequence in {@code right}
      *
+     * @since 2.0
+     */
+    protected void algorithmC(final int leftStart, final int leftEnd, final int rightStart, final int rightEnd) {
+        // Loading arguments from instance fields
+        final CharSequence left = this.leftSequence;
+        final CharSequence right = this.rightSequence;
+        final StringBuilder out = this.algorithmCOut;
+
+        final int leftSz = 1 + leftEnd - leftStart;
+        final int rightSz = 1 + rightEnd - rightStart;
+
+        if (leftSz == 1) { // Handle trivial cases, as per the paper
+            final int leftCh = left.charAt(leftStart);
+            for (int j = 0; j < rightSz; j++) {
+                if (leftCh == right.charAt(rightStart + j)) {
+                    out.appendCodePoint(leftCh);
+                    break;
+                }
+            }
+        } else if (rightSz > 0 && leftSz > 1) {
+            final int mid = leftSz / 2; // Find the middle point
+
+            // Step 3 of the algorithm: two calls to Algorithm B
+            // Parameters for the first call
+            this.leftStart = leftStart;
+            this.leftEnd = leftStart + (mid - 1);
+            this.rightStart = rightStart;
+            this.rightEnd = rightEnd;
+            // this.this.leftSequence contains left input sequence
+            final int[] l1 = algorithmB(this.buffers[0], false);
+            // Parameters for the second call
+            this.leftStart = leftStart + mid;
+            this.leftEnd = leftEnd;
+            this.rightStart = rightStart;
+            this.rightEnd = rightEnd;
+            // this.this.leftSequence contains right input sequence
+            final int[] l2 = algorithmB(this.buffers[1], true);
+
+            // Find k, as per the Step 4 of the algorithm
+            int k = 0;
+            int t = 0;
+            for (int j = 0; j <= rightSz; j++) {
+                final int s = l1[j] + l2[rightSz - j];
+                if (t < s) {
+                    t = s;
+                    k = j;
+                }
+            }
+
+            // Step 5: solve simpler problems, recursively
+            algorithmC(leftStart, leftStart + (mid - 1), rightStart, rightStart + (k - 1));
+            algorithmC(leftStart + mid, leftEnd, rightStart + k, rightEnd);
+        }
+    }
+
+    /**
      * Computes the lcsLengthArray for the sake of doing the actual lcs calculation. This is the
      * dynamic programming portion of the algorithm, and is the reason for the runtime complexity being
      * O(m*n), where m=left.length() and n=right.length().
      *
-     * @param left first character sequence
+     * @param left  first character sequence
      * @param right second character sequence
      * @return lcsLengthArray
+     * @deprecated Deprecated as of 2.0. A more efficient implementation for calculating LCS is now available.
```

Comment:
Not 2.0. Deprecated as of 1.10... And add a note that this method will be removed in 2.0. Similar to `logestCommonSubsequence`'s comments.